### PR TITLE
Use stdout and stderr for logging when in a Docker container

### PR DESCRIPTION
**Pull request by dansomething**
- All Splunk events go to stdout
- All other server (error,info,debug) logs go to stderr
- Add minimist to handle new --debug and --docker switches
- Add morgan to create Apache compatible logs, but sent to stderr

### CODE REVIEW

**Comment by sesteel** [comment]
looks fine

**Comment by joshuawarner32** [comment]
:+1:

**Comment by dansomething** [comment]
FYI, the Dockerfile was previously added in another commit. Its only a couple of lines.

I added the --docker switch to main the original contract in hopes that it will help get this change merged upstream.

File: server.js [hunk 1]
```diff
@@ -86,21 +97,30 @@ var createAndLogEvent = function(data, req) {
...
 };
 
 /*
  * Use Middlewares
  **********************************
  */
-app.use(express.logger());
+if (argv.docker) {
+  var morgan = require('morgan');
+  app.use(morgan('combined', { stream: process.stderr }));
```
Comment (dansomething) [inline server.js:120]: @elementalvoid Can you take a look at this pull request. In the specific change I wasn't sure if we should bother tracking server access logs or not.

Comment (elementalvoid) [inline server.js:120]: What about sending the tracking logs to stdout and server logs to stderr?

Comment (dansomething) [inline server.js:120]: @elementalvoid Yes, exactly. That's what this does. This line is for the server logs. I just wasn't sure if we wanted to capture them. It creates the Apache combined log and sends it to stderr. Up above in the createAndLogEvent method the event is sent to console.info which goes to stdout.

Comment (elementalvoid) [inline server.js:120]: Perfect! Sorry, I'm not at all familiar with the module you were using so I didn't catch that.. Looks good then.

Comment (dansomething) [inline server.js:120]: Cool, so I'll get this merged into master tomorrow which should make it shippable in Docker unless we need to doing anything for SSL support. Right now it just has some dummy certs for testing.

**Pull request merged**